CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ endif()
 
 project( ni-media )
 
+option( NIMEDIA_TREAT_WARNINGS_AS_ERRORS      "Treat compile warnings as errors"      OFF)
```

Comment:
any risk to have it on by default?

Reply:
I think having it ON by default might cause some trouble. I personally don't like it so much to get a compile error for every single warning while developing a new feature. Once I am done i fix the warnings and then i commit. Also I am not sure if every combination of  3rdparty library version and compiler version will compile without warnings. We only test a subset on the CI and i woudn't want people with other settings to have their builds fail because of some minor warning. The main goal here is to not introduce any new warnings when merging back to develop as the CI will now reject it.

Reply:
:) sure thing, so we just check it during the integration build